lib/db/transactions.go

```diff
@@ -453,26 +456,28 @@ func (t readWriteTransaction) close() {
 	t.WriteTransaction.Release()
 }
 
-func (t readWriteTransaction) putFile(key []byte, fi protocol.FileInfo) error {
-	if fi.Blocks != nil {
-		if fi.BlocksHash == nil {
-			fi.BlocksHash = protocol.BlocksHash(fi.Blocks)
-		}
-		blocksKey := t.keyer.GenerateBlockListKey(nil, fi.BlocksHash)
-		if _, err := t.Get(blocksKey); backend.IsNotFound(err) {
+func (t readWriteTransaction) putFile(fkey []byte, fi protocol.FileInfo) error {
+	var bkey []byte
+
+	if len(fi.Blocks) > blocksIndirectionCutoff {
```

Comment:
Does idxchk need handling of this?

Reply:
Not really. It would, if we did your change to always have the hash, but as is now the idxck checks the hash if it's present and otherwise not.